### ISSUE TITLE
`clang-format`: new Dockerfile & update version (fix alignment regression)

### ISSUE THREAD

**Issue by Dominik-K** [EVerest/everest-ci#11]
The `clang-format` version used in the GitHub action  `everest/everest-ci/github-actions/run-clang-format@v1.0.0` is `16.0.0`. This version contains this regression:
- https://github.com/llvm/llvm-project/issues/62161

 which leads to different formatting than what's done locally. E.g. [this current one](https://github.com/EVerest/libocpp/actions/runs/7119319848/job/19384164290?pr=293#step:4:15):
```diff
-inline const std::string MEMORYEXHAUSTION = "MemoryExhaustion"; // CRITICAL
+inline const std::string MEMORYEXHAUSTION = "MemoryExhaustion";           // CRITICAL
```

The regression is `fixed` since version `16.0.2` (see [release-notes](https://discourse.llvm.org/t/llvm-16-0-2-release/70107)).

## Tasks
see https://github.com/EVerest/everest-ci/issues/11#issuecomment-1863029492

**Comment by hikinggrass** [comment]
I agree with @andistorm here, let's try to keep the binary and script local. But nice find, like we suspected

**Comment by hikinggrass** [comment]
The current clang-format-16 binary is uploaded as a release here: https://github.com/EVerest/run-clang-format/releases
So replacing that with the static one should do the trick.

**Comment by hikinggrass** [comment]
In this specific case I disagree that forking wasn't a good idea, and I think some of the things you mentioned are not quite correct

1. If you have a look at https://github.com/EVerest/run-clang-format/releases the binary is purposefully not checked in, just uploaded as a "release" on github, we should do this with the static binary as well. 
2.  I forked the run-clang-format script in anticipation that we might want to change something in this script for our ci, the upstream repo hasn't been updated in over 3 years, so maintenance is basically on us anyway if we want to fix anything.
3.  I don't want to break the current docker image that we told people to use for months now, without a good reason. And in that case we *must* update the binary, otherwise it will stay on the broken v16 release forever

So in this case I think there have been very valid reasons to do this in this way, but I totally agree that I should have documented it somewhere :sweat_smile: I guess this is what happens when you hack away on these sort of things in a boiling tent on the CCCamp :laughing:

**Comment by Dominik-K** [comment]
We've had a meeting on Dec, 8th, to discuss this. Our outcome was:

* `everest-utils/docker` to be moved to `everest-ci`
* reusable workflows
  - Actions in `everest-ci/github-actions`
  - composite actions
* GitHub Actions workflow files with few magic (as possible)
* `run-clang-format.py` next to Dockerfile
   - archive or remove `run-clang-format` repo
* `clang-format` binary as release artifact
   - automatically downloaded via CI [from `llvm-project`](https://github.com/llvm/llvm-project/releases/tag/llvmorg-15.0.6) and then cached on our side
    - `clang-format` version 15.0.6 because Ubuntu 22.04 LTS offers `clang-format-15` as highest version

Since then we had another meeting which got into detail of some of the topics and overhauling some ideas. Meeting notes might be added soon.

**Comment by andistorm** [comment]
Fixed by downgrading clang-format to 15.0.6
